src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -158,6 +157,8 @@ public void Initialize(int streamId, int initialWindowSize)
 
             public Http2Connection Connection => _connection;
 
+            public bool WebSocketEstablished { get; private set; }
```

Comment:
Is all of this more general than websockets? Should all of this websocket naming instead reflect general usage of a connect header, which could be used for protocols besides websockets? Add we've discussed, we're trying to keep SocketsHttpHandler ignorant of websockets, imbuing it with whatever support it needs for such protocols to be implemented on top.

Reply:
Renamed to `ConnectProtocolEstablished`